tests/usbprotocol/test_comms.py

```diff
@@ -99,31 +102,30 @@ def test_comms_reset_tx(self):
     for m in queue_msgs:
       assert m == test_msg, "message buffer should contain valid test messages"
 
-
-  def test_can_send_usb(self):
+  @pytest.mark.parametrize("bus", range(3))
+  @pytest.mark.order(2)
+  def test_can_send_usb(self, bus):
     lpp.set_safety_hooks(Panda.SAFETY_ALLOUTPUT, 0)
+    for _ in range(100):
+      msgs = random_can_messages(200, bus=bus)
+      packed = pack_can_buffer(msgs)
 
-    for bus in range(3):
-      with self.subTest(bus=bus):
-        for _ in range(100):
-          msgs = random_can_messages(200, bus=bus)
-          packed = pack_can_buffer(msgs)
-
-          # Simulate USB bulk chunks
-          for buf in packed:
-            for i in range(0, len(buf), CHUNK_SIZE):
-              chunk_len = min(CHUNK_SIZE, len(buf) - i)
-              lpp.comms_can_write(buf[i:i+chunk_len], chunk_len)
+      # Simulate USB bulk chunks
+      for buf in packed:
+        for i in range(0, len(buf), CHUNK_SIZE):
+          chunk_len = min(CHUNK_SIZE, len(buf) - i)
+          lpp.comms_can_write(buf[i:i+chunk_len], chunk_len)
 
-          # Check that they ended up in the right buffers
-          queue_msgs = []
-          pkt = libpanda_py.ffi.new('CANPacket_t *')
-          while lpp.can_pop(TX_QUEUES[bus], pkt):
-            queue_msgs.append(unpackage_can_msg(pkt))
+      # Check that they ended up in the right buffers
+      queue_msgs = []
+      pkt = libpanda_py.ffi.new('CANPacket_t *')
+      while lpp.can_pop(TX_QUEUES[bus], pkt):
+        queue_msgs.append(unpackage_can_msg(pkt))
 
-          self.assertEqual(len(queue_msgs), len(msgs))
-          self.assertEqual(queue_msgs, msgs)
+      assert len(queue_msgs) == len(msgs)
+      assert queue_msgs == msgs
 
+  @pytest.mark.order(1)
```

Comment:
why is this needed? the tests should be independent

Reply:
@adeebshihadeh I have explained this in PR description The tests are dependent on a definite ordering which be reproduced with unittests if you change the method names. E.g test_zzz... for the order(5) five test for example.

Reply:
@adeebshihadeh just bumping this since I haven't heard anything back in a while. I think the easiest way to allow parallelism would be to mark the tests that depend on a particular order to run serially. that way parallelism could still be used for other tests. Fixing the tests themselves, seems like a more complicated task.